### DISCUSSION TITLE
Docker run command needs port mapping...but closes the connection

### DISCUSSION THREAD

**Discussion by markingle** [tsightler/ring-mqtt#959] [General Discussion]
I was having a problem getting the docker instance to make a connection to my local mosquitto service so I added the "-p 1883:1883" mapping option to the run command.  This helped but now I am getting a connect/disconnect msg in the mosquitto logs.  Are these any logs on the ring-mqtt side of things that can help me troubleshoot?

![image](https://github.com/user-attachments/assets/2eda4849-9536-4d03-a714-1c50310c7fbe)

![image](https://github.com/user-attachments/assets/dab17a01-931c-44d7-95a2-16a0681bf42b)

**Comment by tsightler** [comment]
I have no idea what you are trying to do since you posted just screenshots of logs with no details of your actual config, i.e. how are you running ring-mqtt, where is your mqtt broker running, etc.  the fact that it's showing connections from 127.0.0.1 already indicates that something is wrong since that should never be the case if you are running ring-mqtt via Docker.  Is your broker also in Docker or also running on the host?

In your example above it appears you are running mosquitto on the host, in local only mode, but running ring-mqtt in Docker.  This is not a valid configuration, unless you also run ring-mqtt using the host network.  There should be no reason to map ports in Docker to access the broker.

**Reply by tsightler** [reply]
Oops, never mind, I originally only saw the second screenshot.   You are using the MQTT url mqtt://localhost:1883 while running in docker, this will never be a correct config unless you run ring-mqtt on the host network.  You should put the IP address of the host running mosquitto there, not localhost.  No reason to be mapping any ports, unless you are also running mosquitto in a separate Docker, then you can also use the container name and the connection will occur over the Docker network.  However, critical is this message:

"Starting in local only mode.  Connections will only be possible from clients running on this machine."

Docker is not "this machine" unless you run Docker with -net=host (not really recommended, but certainly possible).  You should configure Mosquitto to listen for connections on some address other than localhost, either the host itself, or the Docker netework.

**Comment by markingle** [comment]
I agree that I should not need the port mapping.  I am to get just the basics going so that I can understand what is going on with your project. Does the docker container include a broker?

Here is my config.json file...

"mqtt_url": "mqtt://localhost:1883",
    "mqtt_options": "",
    "livestream_user": "",
    "livestream_pass": "",
    "disarm_code": "",
    "enable_cameras": true,
    "enable_modes": false,
    "enable_panic": false,
    "hass_topic": "homeassistant/status",
    "ring_topic": "ring",
    "location_ids": []

**Reply by tsightler** [reply]
Sorry, see my second reply above (I initially missed your first screenshot).  The ring-mqtt docker container does not have a publicly accessible broker, you must have an external broker which is reachable from the Docker container.  Mosquitto running on the host is fine, but it can't just listen on localhost since localhost in the container is different than localhost on the host itself (alternatively, you can run Docker on host network, but I don't recommend this for security reasons, although it is possible).  You must configre Mosquitto to listen on the host IP address, or at least the Docker network IP, and then point at that, not localhost.  Localhost for the container is the internal localhost, nothing is listening there.

**Reply by markingle** [reply]
Wow...thanks!  Yeah I get confused with docker solutions and forget that (virtually) the container is a separate system all together!  Once I got the IP address corrected in the config.json file and updated the mosquitto.conf with the correct listener and anonymous settings things took off!

![image](https://github.com/user-attachments/assets/c28ee896-00b5-4442-ba15-08c2e51cd1ec)
